File: 252.meeting-rooms.java
import java.lang.reflect.Array;

/*
 * @lc app=leetcode id=252 lang=java
 *
 * [252] Meeting Rooms
 */

// @lc code=start

class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return a[0] - b[0];
            }
        });

        for (int i = 1; i < intervals.length; i++) {
            if (intervals[i - 1][1] > intervals[i][0]) {
                return false;
            }

        }

        return true;
    }
}
// @lc code=end

File: 252.meeting-rooms.py
#
# @lc app=leetcode id=252 lang=python3
#
# [252] Meeting Rooms
#

# @lc code=start

class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        n=len(intervals)
        intervals.sort(key=lambda interval:interval[0])
        for i in range(1,n):
            if intervals[i][0]<intervals[i-1][1]:
                return False

        return True

# @lc code=end

